// Определение версии синтаксиса ProtoBuf
syntax = "proto3";

// Опционально: указываем пакет
package irc;

// import "google/protobuf/timestamp.proto";


service IrcService {
  rpc GetToken(IrcVoid) returns (IrcToken);
  rpc SendMessage(IrcMessage) returns (IrcVoid);
  rpc GetMessages(IrcToken) returns (stream IrcMessage);
}

message IrcToken{
    string token = 1;
}

message IrcMessage {
  string message = 1;
  string token = 2;
}

//message IrcReply {
//  string message = 1;
//}

message IrcVoid {

}



